api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/calculation/pvls/InitialArtStartDateCalculationTest.java

```diff
@@ -0,0 +1,125 @@
+            evaluationContext));
+    // initiated ART by ARV transfer in observation
+    map.put(
+        999,
+        new SimpleResult(
+            new Timestamp(testsHelper.getDate("2019-01-20 00:00:00.0").getTime()),
+            calculation,
+            evaluationContext));
+    map.put(432, new SimpleResult(null, calculation, evaluationContext));
+    return map;
+  }
+
+  @Test
+  public void evaluateShouldConsiderLeastInitiationDateForMultipleEntryPoints() {
+    // initiating ART by starting ARV plan observation in addition to HIV
+    // enrollment
+    PatientCalculationContext evaluationContext = getEvaluationContext();
+    openmrsTestHelper.createBasicObs(
+        Context.getPatientService().getPatient(2),
+        Context.getConceptService().getConcept(7777002),
+        Context.getEncounterService().getEncounter(3),
+        testsHelper.getDate("2008-07-01 00:00:00.0"),
+        (Location) evaluationContext.getFromCache("location"),
+        Context.getConceptService().getConcept(7777003));
+    CalculationResultMap evaluatedResult =
+        service.evaluate(getCohort(), getCalculation(), evaluationContext);
+    Assert.assertEquals(
+        testsHelper.getDate("2008-07-01 00:00:00.0"), evaluatedResult.get(2).getValue());
+    // the rest should not be touched
+    matchOtherResultsExcept(evaluatedResult, 2);
+
+    // initiating ART by historical start date in addition to HIV enrollment
+    openmrsTestHelper.createBasicObs(
+        Context.getPatientService().getPatient(2),
+        Context.getConceptService().getConcept(7777005),
+        Context.getEncounterService().getEncounter(3),
+        testsHelper.getDate("2019-01-01 00:00:00.0"),
+        (Location) evaluationContext.getFromCache("location"),
+        testsHelper.getDate("2008-06-01 00:00:00.0"));
+    evaluatedResult = service.evaluate(getCohort(), getCalculation(), evaluationContext);
+    Assert.assertEquals(
+        testsHelper.getDate("2008-06-01 00:00:00.0"), evaluatedResult.get(2).getValue());
+    // the rest should not be touched
+    matchOtherResultsExcept(evaluatedResult, 2);
+
+    // initiating ART by historical start date in addition to HIV enrollment
+    openmrsTestHelper.createBasicObs(
+        Context.getPatientService().getPatient(2),
+        Context.getConceptService().getConcept(7777002),
+        Context.getEncounterService().getEncounter(3),
+        testsHelper.getDate("2019-01-02 00:00:00.0"),
+        (Location) evaluationContext.getFromCache("location"),
+        Context.getConceptService().getConcept(7777004));
+    evaluatedResult = service.evaluate(getCohort(), getCalculation(), evaluationContext);
+    Assert.assertEquals(
+        testsHelper.getDate("2008-06-01 00:00:00.0"), evaluatedResult.get(2).getValue());
+    // the rest should not be touched
+    matchOtherResultsExcept(evaluatedResult, 2);
+
+    // TODO does the results have to contain every patient from the cohort
+    // received with empty values!
```

Comment:
What do you mean here?

InitialArtStartDateCalculation should return a date value for each patient if they are enrolled on ARV and if not then just return null.